File: AkitaModel/Results/VisitSale.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Skyware.Lis.AkitaModel.Results;

public class VisitSale
{

    /// <summary>
    /// PK in LIS iLab.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Date/time the sale was created.
    /// </summary>
    public DateTime Date { get; set; }

    /// <summary>
    /// List of billable products.
    /// </summary>
    public IList<Product> Products { get; set; }

    /// <summary>
    /// List of payments for this sale.
    /// </summary>
    public IList<PaymentBase> Payments { get; set; }

    /// <summary>
    /// Total price of the sale.
    /// </summary>
    public decimal Total => Products?.Sum(p => p.Price) ?? 0m;

    /// <summary>
    /// Balance of the sale.
    /// </summary>
    public decimal Balance => Total - (Payments?.Sum(p => p.Amount) ?? 0m);

}

File: AkitaMotelTS/src/Results/VisitSale.ts
/**
 * Skyware.Lis.AkitaModel.Results
 */

import { PaymentBase } from '../PaymentBase';
import { Product } from './Product';

export class VisitSale {
  /**
   * PK in LIS iLab.
   */
  id!: number;

  /**
   * Date/time the sale was created.
   */
  date!: Date;

  /**
   * List of billable products.
   */
  products?: Product[];

  /**
   * List of payments for this sale.
   */
  payments?: PaymentBase[];

  /**
   * Total price of the sale.
   */
  get total(): number {
    return this.products?.reduce((sum, p) => sum + p.price, 0) ?? 0;
  }

  /**
   * Balance of the sale.
   */
  get balance(): number {
    return this.total - (this.payments?.reduce((sum, p) => sum + p.amount, 0) ?? 0);
  }
}
